### DISCUSSION TITLE
Qns on the usage of useFieldArray, useFormContext, react-select and their integration

### DISCUSSION THREAD

**Discussion by this9is3me** [react-hook-form/react-hook-form#4207] [Q&A]
Hi @bluebill1049 firstly, thanks for developing this react-hook form!
I have been trying out the usage of useFieldArray, useFormContext and wrapping React-select with Controller.
[Codesandbox link](https://codesandbox.io/s/react-hook-form-v6-controller-append-prepepend-insert-forked-utit6?fontsize=14&hidenavigation=1&theme=dark&file=/src/index.js:785-798)

I am trying to create rows which contain input and React-select dropdown in each row. To create a row, click on append button.

However, I have faced some issues faced and really need your help to clarify the following:
1. Am I using the usefieldarray and controller correctly?
2. How do I retrieve the latest values from the input and react-select dropdown after clicking append button?
3. How do I make the dropdown showing the value being selected?

Sorry for these noob questions.
Really appreciate your answers for the above.

**Comment by bluebill1049** [comment]
looks like the only thing you are missing out is the rende props:

```
        render={(prop) => {
          return (
            <ReactSelect
              id={`dropdown${data.id}`}
              name={`dropdown${data.id}`}
              initialValue={data.dropdown}
              dropdownMenu={[
                { value: "1", label: "1" },
                { value: "2", label: "2" },
                { value: "3", label: "3" },
                { value: "4", label: "4" }
              ]}
              {...prop}
            />
          );
        }}
```

**Reply by this9is3me** [reply]
hmm after add the props in, the dropdown is working now but console is throwing warning:
![image](https://user-images.githubusercontent.com/6081672/108026281-03a2b480-7063-11eb-9d7c-32ed24df568f.png)

**Reply by this9is3me** [reply]
also wondering for this part:
`control, // control props comes from useForm (optional: if you are using FormContext)`
1. optional means no need to include this control in useFieldArray?

2. Another qns is I have another usage of control from useFormContext().
const { control } = useForm();
const { control } = useFormContext();
Is there a difference between these 2 control?

**Reply by bluebill1049** [reply]
> also wondering for this part:
> `control, // control props comes from useForm (optional: if you are using FormContext)`
> 
> 1. optional means no need to include this control in useFieldArray?
> 2. Another qns is I have another usage of control from useFormContext().
>    const { control } = useForm();
>    const { control } = useFormContext();
>    Is there a difference between these 2 control?

1. yes, if you are using formContext
2. same, useFormContext is when you need to get access for control at deep nested compoennt.

**Reply by this9is3me** [reply]
> hmm after add the props in, the dropdown is working now but console is throwing warning:
> ![image](https://user-images.githubusercontent.com/6081672/108026281-03a2b480-7063-11eb-9d7c-32ed24df568f.png)

then for this error^ this happens after i added the props into the render for controller

**Reply by bluebill1049** [reply]
have a read in the doc for controller: https://react-hook-form.com/api#Controller and see what' render is passing down. if ref is not available don't assign it.

**Reply by this9is3me** [reply]
sorry don't really understand where these props are passing from...
so this render prop is generated by the Controller itself?

**Reply by bluebill1049** [reply]
`({ ref, ...rest })` you can omit the ref so they don't get assigned
